src/openclsim/critical_path/base_cp.py

```diff
@@ -115,6 +126,40 @@ def combine_logs(self):
 
         return log_all.sort_values("Timestamp").reset_index(drop=True)
 
+    @staticmethod
+    def get_log_dataframe_activity(activity_list):
+        """
+        Get the log of the activity object in a pandas dataframe.
+
+        Parameters
+        ----------
+        activity_list : list
+            list of recorded activities, i.e. every activity has log attribute.
+        """
+
+        id_map = {act.id: act.name for act in activity_list}
+
+        all_dfs_list = []
+        for sub_activity in activity_list:
+            all_dfs_list.append(sub_activity.log)
+
+        all_timestamps = [ts for dict_ in all_dfs_list for ts in dict_["Timestamp"]]
+        all_activity_id = [id_ for dict_ in all_dfs_list for id_ in dict_["ActivityID"]]
+        all_states = [
+            state for dict_ in all_dfs_list for state in dict_["ActivityState"]
+        ]
+        df_all = pd.DataFrame(
```

Comment:
tsja probleem. We leunen inderdaad nu op de huidige naamgeving.

Reply:
@Pietervanhalem Maybe interesting to discuss in next stakeholder session? (Especially since Fedor's changes in logging might fit/discourage/encourage this practice or a new practice.)